format-delimited/src/test/java/io/cdap/plugin/format/delimited/input/DelimitedStringTest.java

```diff
@@ -0,0 +1,107 @@
+ * the License.
+ */
+
+package io.cdap.plugin.format.delimited.input;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class DelimitedStringTest {
+
+  @Test
+  public void testStringShorterThanDelimiter() throws Exception {
+    String test = "a";
+    Assert.assertEquals(Arrays.asList(test.split(",,")),
+        PathTrackingDelimitedInputFormat.splitQuotesString(test, ",,"));
+    Assert.assertEquals(Arrays.asList(test.split("aa")),
+        PathTrackingDelimitedInputFormat.splitQuotesString(test, "aa"));
+  }
+
+  @Test
+  public void testStringWithConsecutiveDelimiter() throws Exception {
+    String test = "aaa";
+    Assert.assertEquals(Arrays.asList(test.split("a", -1)),
+        PathTrackingDelimitedInputFormat.splitQuotesString(test, "a"));
+    Assert.assertEquals(Arrays.asList(test.split("aa", -1)),
+        PathTrackingDelimitedInputFormat.splitQuotesString(test, "aa"));
+
+    test = "aaaaaaa";
+    Assert.assertEquals(Arrays.asList(test.split("aa", -1)),
+        PathTrackingDelimitedInputFormat.splitQuotesString(test, "aa"));
+    test = "aaaaaaaa";
+    Assert.assertEquals(Arrays.asList(test.split("aa", -1)),
+        PathTrackingDelimitedInputFormat.splitQuotesString(test, "aa"));
+  }
+
+  @Test
+  public void testSimpleSplit() throws Exception {
+    String test = "a,b,c,d,e";
+    Assert.assertEquals(Arrays.asList(test.split(",")),
+        PathTrackingDelimitedInputFormat.splitQuotesString(test, ","));
+
+    test = "a1,b1,c1,d1,e1";
+    Assert.assertEquals(Arrays.asList(test.split(",")),
+        PathTrackingDelimitedInputFormat.splitQuotesString(test, ","));
+
+    test = "1###sam###a@b.com###male";
+    Assert.assertEquals(Arrays.asList(test.split("###")),
+        PathTrackingDelimitedInputFormat.splitQuotesString(test, "###"));
+
+    test = "a1,,,b1,,,c1,,,d1,,,e1";
+    Assert.assertEquals(Arrays.asList(test.split(",")),
+        PathTrackingDelimitedInputFormat.splitQuotesString(test, ","));
+  }
+
+  @Test
+  public void testValidQuotesWithTrimedQuotes() {
+    String test = "a,\"b,c\"";
+    List<String> expected = ImmutableList.of("a", "b,c");
+    Assert.assertEquals(expected, PathTrackingDelimitedInputFormat.splitQuotesString(test, ","));
+
+    test = "a,\"aaaaa\"aaaab\"aaaac\",c";
+    expected = ImmutableList.of("a", "aaaaa\"aaaab\"aaaac", "c");
+    Assert.assertEquals(expected, PathTrackingDelimitedInputFormat.splitQuotesString(test, ","));
+
+    test = "a,\"b,c\",\"d,e,f\"";
+    expected = ImmutableList.of("a", "b,c", "d,e,f");
+    Assert.assertEquals(expected, PathTrackingDelimitedInputFormat.splitQuotesString(test, ","));
+
+    test = "a###\"b###c\"###\"d###e###f\"";
+    expected = ImmutableList.of("a", "b###c", "d###e###f");
+    Assert.assertEquals(expected, PathTrackingDelimitedInputFormat.splitQuotesString(test, "###"));
+  }
+
+  @Test
+  public void testBadQuotes() {
```

Comment:
this test should never pass as no exception is thrown in the method. Should make sure the added unit tests pass before commit